### PR TITLE
[Android] Don't consider display mode ids constant.

### PR DESCRIPTION
**Pull request by ksooo**
Display mode ids may change at any time, for instance after an HDMI reset. Not taking this into account makes switching native resolution fail in case mode ids have changed. 

@peak3d like discussed on Slack...

### CODE REVIEW

File: xbmc/windowing/android/AndroidUtils.h [hunk 1]
```diff
@@ -24,6 +24,7 @@ class CAndroidUtils : public ISettingCallback
   virtual bool  GetNativeResolution(RESOLUTION_INFO *res) const;
   virtual bool  SetNativeResolution(const RESOLUTION_INFO &res);
   virtual bool  ProbeResolutions(std::vector<RESOLUTION_INFO> &resolutions);
+  virtual bool  UpdateDisplayModes();
```
Comment (Rechi) [inline xbmc/windowing/android/AndroidUtils.h]: one space between `bool` and `UpdateDisplayModes` is enough

File: xbmc/windowing/android/WinSystemAndroid.h [hunk 2]
```diff
@@ -76,4 +78,6 @@ class CWinSystemAndroid : public CWinSystemBase, public ITimerCallback
   CCriticalSection m_resourceSection;
   std::vector<IDispResource*> m_resources;
   CDecoderFilterManager *m_decoderFilterManager;
+private:
```
Comment (Rechi) [inline xbmc/windowing/android/WinSystemAndroid.h]: add an empty above this line

File: xbmc/settings/DisplaySettings.cpp [hunk 3]
```diff
@@ -219,6 +219,7 @@ void CDisplaySettings::Clear()
   CSingleLock lock(m_critical);
   m_calibrations.clear();
   m_resolutions.clear();
+  m_resolutions.insert(m_resolutions.begin(), RES_CUSTOM, RESOLUTION_INFO());
```
Comment (notspiff) [inline xbmc/settings/DisplaySettings.cpp]: can be a push_back.

Comment (ksooo) [reply xbmc/settings/DisplaySettings.cpp]: Yep, thanks.

**Comment by ksooo** [comment]
@Rechi @notspiff thanks for the feedback. PR is updated now.

File: xbmc/settings/DisplaySettings.cpp (hunk 3 above)
Comment (peak3d) [reply xbmc/settings/DisplaySettings.cpp]: I dont agree, this is filling all entries until RES_DESKTOP (exclusive) with an empty resolution

Comment (ksooo) [reply xbmc/settings/DisplaySettings.cpp]: Sure, this is the intention. Erasing all elements and filling the vector with RES_CUSTOM elements. This is, what the ctor of this class does and was forgotten here. If CDisplaySettings::Clear() gets called and after this CDisplaySettings::AddResolutionInfo() which is supposed to add custom resos which must inserted into the vector at index RES_CUSTOM + n would not work.

**Comment by ksooo** [comment]
@peak3d requested changes are in.

**Review by peak3d — APPROVED** [review]

**Pull request merged**